fdbserver/DataDistributionQueue.actor.cpp

```diff
@@ -1009,12 +1056,22 @@ struct DDQueueData {
 			for (int r = 0; r < ranges.size(); r++) {
 				RelocateData& rrs = inFlight.rangeContaining(ranges[r].begin)->value();
 				rrs.keys = ranges[r];
+				if (rd.keys == ranges[r] && rd.isRestore()) {
+					ASSERT(rd.dataMove != nullptr);
+					rrs.dataMoveId = rd.dataMove->meta.id;
+				} else {
+					// TODO(psm): The shard id is determined by DD.
```

Comment:
For the case: rd.keys != ranges[r] && rd.isRestore(), why do we create a new dataMoveId rather than using the dataMove->meta.id? Could you add some comments on this?